### PR TITLE
Fix: BingoCardTips IndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by Thunderblade73**
## Discord
https://discord.com/channels/997079228510117908/1193356023432495134

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/bingo/card/BingoCardTips.kt [hunk 1]
```diff
@@ -6,22 +6,32 @@ import at.hannibal2.skyhanni.events.LorenzToolTipEvent
...
 
 class BingoCardTips {
+
+    private val repoGroup = RepoPattern.group("bingo.card.tips")
+
     private val config get() = SkyHanniMod.feature.event.bingo.bingoCard
 
+    private val inventoryPattern by repoGroup.pattern("card", "Bingo Card")
+    private val rewardPattern by repoGroup.pattern("reward", "§.§.§7Reward")
+    private val contributionRewardsPattern by repoGroup.pattern("reward.contribution", "§.§.§7Contribution Rewards.*")
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/bingo/card/BingoCardTips.kt]:
```suggestion
    private val config get() = SkyHanniMod.feature.event.bingo.bingoCard

    private val patternGroup = RepoPattern.group("bingo.card.tips")
    private val inventoryPattern by patternGroup.pattern(
        "card", 
        "Bingo Card"
    )
    private val rewardPattern by patternGroup.pattern(
        "reward", 
        "§.§.§7Reward"
    )
    private val contributionRewardsPattern by patternGroup.pattern(
        "reward.contribution", 
        "§.§.§7Contribution Rewards.*"
    )
```

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
just this one thing to match the style of https://github.com/hannibal002/SkyHanni/pull/871 better and also improves readability of the patterns

File: src/main/java/at/hannibal2/skyhanni/features/bingo/card/BingoCardTips.kt [hunk 2]
```diff
@@ -19,10 +22,24 @@ class BingoCardTips {
...
     private val config get() = SkyHanniMod.feature.event.bingo.bingoCard
 
+    private val patternGroup = RepoPattern.group("bingo.card.tips")
+    private val inventoryPattern by patternGroup.pattern(
+        "card", 
+        "Bingo Card"
+    )
+    private val rewardPattern by patternGroup.pattern(
+        "reward", 
+        "§.§.§7Reward"
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/bingo/card/BingoCardTips.kt:32]: Technically this string always has to start with `§5§o` as this is a minecraft limitation, not controlled by hypixel.

File: src/main/java/at/hannibal2/skyhanni/features/bingo/card/BingoCardTips.kt [hunk 3]
```diff
@@ -19,10 +22,24 @@ class BingoCardTips {
 
     private val config get() = SkyHanniMod.feature.event.bingo.bingoCard
 
+    private val patternGroup = RepoPattern.group("bingo.card.tips")
+    private val inventoryPattern by patternGroup.pattern(
+        "card",
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/bingo/card/BingoCardTips.kt]: rename to "card title" or something

**Review by hannibal002 — APPROVED** [review]
works fine any way

**Comment by hannibal002** [comment]
Dont ask please

**Pull request merged**